### PR TITLE
Further Bfme ini parsing

### PR DESCRIPTION
**Pull request by Tarcontar**

### CODE REVIEW

File: src/OpenSage.Game/Logic/Object/Draw/W3dHordeModelDraw.cs [hunk 1]
```diff
@@ -9,6 +10,39 @@ public class W3dHordeModelDrawModuleData : W3dModelDrawModuleData
...
+            {
+                { "StaticModelLODMode", (parser, x) => x.StaticModelLODMode = parser.ParseBoolean() },
+                { "LodOptions", (parser, x) => x.LODOptions.Add(LODOption.Parse(parser)) }
+            });
+
+        public bool StaticModelLODMode { get; internal set; }
+        public List<LODOption> LODOptions { get; internal set; } = new List<LODOption>();
+    }
+
+    public sealed class LODOption
```
Comment (tgjones) [inline src/OpenSage.Game/Logic/Object/Draw/W3dHordeModelDraw.cs]: Could you rename this to `LodOption`, to match .NET naming convention that acronyms with three letters or more should be Pascal-cased?

File: src/OpenSage.Game/Logic/Object/Draw/W3dHordeModelDraw.cs [hunk 2]
```diff
@@ -9,6 +10,39 @@ public class W3dHordeModelDrawModuleData : W3dModelDrawModuleData
...
         internal static new readonly IniParseTable<W3dHordeModelDrawModuleData> FieldParseTable = W3dModelDrawModuleData.FieldParseTable
-            .Concat(new IniParseTable<W3dHordeModelDrawModuleData>());
+            .Concat(new IniParseTable<W3dHordeModelDrawModuleData>
+            {
+                { "StaticModelLODMode", (parser, x) => x.StaticModelLODMode = parser.ParseBoolean() },
+                { "LodOptions", (parser, x) => x.LODOptions.Add(LODOption.Parse(parser)) }
+            });
+
+        public bool StaticModelLODMode { get; internal set; }
+        public List<LODOption> LODOptions { get; internal set; } = new List<LODOption>();
```
Comment (tgjones) [inline src/OpenSage.Game/Logic/Object/Draw/W3dHordeModelDraw.cs]: Please rename this to `LodOptions`.

File: src/OpenSage.Game/Logic/Object/Draw/W3dHordeModelDraw.cs [hunk 3]
```diff
@@ -9,6 +10,39 @@ public class W3dHordeModelDrawModuleData : W3dModelDrawModuleData
...
+
+        internal static readonly IniParseTable<LODOption> FieldParseTable = new IniParseTable<LODOption>
+            {
+                { "AllowMultipleModels", (parser, x) => x.AllowMultipleModels = parser.ParseBoolean() },
+                { "MaxRandomTextures", (parser, x) => x.MaxRandomTextures = parser.ParseInteger() },
+                { "MaxRandomAnimations", (parser, x) => x.MaxRandomAnimations = parser.ParseInteger() },
+                { "MaxAnimFrameDelta", (parser, x) => x.MaxAnimFrameDelta = parser.ParseInteger() }
+            };
+
+        public ModelLevelOfDetail LOD { get; internal set; }
```
Comment (tgjones) [inline src/OpenSage.Game/Logic/Object/Draw/W3dHordeModelDraw.cs]: Please rename this to `Lod`.

File: src/OpenSage.Game/Logic/Object/Draw/W3dHordeModelDraw.cs [hunk 4]
```diff
@@ -9,6 +10,39 @@ public class W3dHordeModelDrawModuleData : W3dModelDrawModuleData
...
+
+        public bool AllowMultipleModels { get; internal set; }
```
Comment (tgjones) [inline src/OpenSage.Game/Logic/Object/Draw/W3dHordeModelDraw.cs]: Do these need to be `internal set`? Could they be `private set`? I've noticed that in a few other places too.

File: src/OpenSage.Game/Logic/Object/SpecialPower/SiegeDepolySpecialPower.cs [hunk 5]
```diff
@@ -0,0 +1,30 @@
+﻿using System.Collections.Generic;
```
Comment (tgjones) [inline src/OpenSage.Game/Logic/Object/SpecialPower/SiegeDepolySpecialPower.cs]: This file's filename has a typo.

File: src/OpenSage.Game/Logic/Object/Draw/W3dHordeModelDraw.cs (hunk 4 above)
Comment (Tarcontar) [reply src/OpenSage.Game/Logic/Object/Draw/W3dHordeModelDraw.cs]: Saw that in a few places, and used internal by default for new objects. Should I use private instead?

Comment (tgjones) [reply src/OpenSage.Game/Logic/Object/Draw/W3dHordeModelDraw.cs]: I thought I only used `internal` when it actually needed to be accessed by something else, like a subclass or another class. Yes, `private` is better if it doesn't need to be accessed by anything else.

Comment (Tarcontar) [reply src/OpenSage.Game/Logic/Object/Draw/W3dHordeModelDraw.cs]: ok, so should i change that for all the objects, also for the ones from the old PR?

Comment (tgjones) [reply src/OpenSage.Game/Logic/Object/Draw/W3dHordeModelDraw.cs]: If you have the energy to do so, yes please.

**Comment by Tarcontar** [comment]
done, anything else?

**Comment by tgjones** [comment]
Thank you!

**Pull request merged**